bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/EscalationEventDefinitionValidator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.model.bpmn.validation.zeebe;
+
+import io.camunda.zeebe.model.bpmn.instance.BoundaryEvent;
+import io.camunda.zeebe.model.bpmn.instance.EscalationEventDefinition;
+import org.camunda.bpm.model.xml.instance.ModelElementInstance;
+import org.camunda.bpm.model.xml.validation.ModelElementValidator;
+import org.camunda.bpm.model.xml.validation.ValidationResultCollector;
+
+public class EscalationEventDefinitionValidator
+    implements ModelElementValidator<EscalationEventDefinition> {
+
+  @Override
+  public Class<EscalationEventDefinition> getElementType() {
+    return EscalationEventDefinition.class;
+  }
+
+  @Override
+  public void validate(
+      final EscalationEventDefinition element,
+      final ValidationResultCollector validationResultCollector) {
+
+    if (!isEscalationBoundaryEvent(element) && element.getEscalation() == null) {
+      validationResultCollector.addError(0, "Must reference an escalation");
+    }
```

Comment:
❌ This validation is incomplete. Currently you are checking that an escalation is present for all escalation events, except for the boundary events. In reality the escalation is optional for all escalation catch events. This means this is for the boundary event and for the event sub process start event.